### PR TITLE
ROX-26318: create a code generation tool for custom resource

### PR DESCRIPTION
**Pull request by c-du**
### Description

change me!

### User-facing documentation

- [ ] CHANGELOG is updated **OR** update is not needed
- [ ] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [ ] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [ ] CI results are inspected

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests

#### How I validated my change

change me!

### CODE REVIEW

File: generated/api/v1/alert_service.swagger.json [hunk 1]
```diff
@@ -1451,16 +1451,13 @@
           }
         },
```
Comment (c-du) [inline generated/api/v1/alert_service.swagger.json:1452]: Please ignore these changes. These changes are from a bug in swagger generation process

**Comment by kylape** [comment]
I kind of think we should go ahead and integrate this code generator into our makefile and add a CI job to ensure it's been run.  wdyt?

**Comment by c-du** [comment]
> I kind of think we should go ahead and integrate this code generator into our makefile and add a CI job to ensure it's been run. wdyt?

This code generator is invoked by `make go-generated-srcs` in customresources.

**Pull request merged**